lib/editor/components/timetable/TripSeriesModal.js

```diff
@@ -17,57 +45,31 @@ type Props = {
 }
 
 const TripSeriesModal = (props: Props) => {
-  // Trip series timing state variables
-  const [startTime, setStartTime] = useState(null)
-  const [headway, setHeadway] = useState(null)
-  const [endTime, setEndTime] = useState(null)
-
-  // Automatic block and trip ID state variables
-  const [useAutoTripIds, setUseAutoTripIds] = useState(false)
-  const [useAutoBlockIds, setUseAutoBlockIds] = useState(false)
-
-  const [autoTripIdStart, setAutoTripIdStart] = useState(0)
-  const [autoBlockIdStart, setAutoBlockIdStart] = useState(1)
-
-  const [autoTripIdIncrement, setAutoTripIdIncrement] = useState(1)
-  const [autoBlockIdIncrement, setAutoBlockIdIncrement] = useState(1)
-
-  const [autoTripIdPrefix, setAutoTripIdPrefix] = useState('')
-  const [autoBlockIdPrefix, setAutoBlockIdPrefix] = useState('')
+  const [state, setState] = useState<State>(initialState)
 
   const messages = getComponentMessages('TripSeriesModal')
 
-  const handleIncrementStartUpdate = (evt: SyntheticInputEvent<HTMLInputElement>) => setAutoTripIdStart(+evt.target.value)
+  const handleIncrementStartUpdate = (evt: SyntheticInputEvent<HTMLInputElement>) => setState((prevState) => ({ ...prevState, autoTripIdStart: +evt.target.value }))
   const handleBlockIdStartUpdate = (evt: SyntheticInputEvent<HTMLInputElement>) => {
     // Check the input to make sure it's within the valid range.
     let input = +evt.target.value
     if (isNaN(input)) return null
     if (input > autoBlockIdIncrement) input = autoBlockIdIncrement
     else if (input < 1) input = 1
-    setAutoBlockIdStart(input)
+    setState((prevState) => ({ ...prevState, autoBlockIdStart: input }))
   }
 
-  const handleTripPrefixUpdate = (evt: SyntheticInputEvent<HTMLInputElement>) => setAutoTripIdPrefix(evt.target.value)
-  const handleBlockPrefixUpdate = (evt: SyntheticInputEvent<HTMLInputElement>) => setAutoBlockIdPrefix(evt.target.value)
+  const handleTripPrefixUpdate = (evt: SyntheticInputEvent<HTMLInputElement>) => setState((prevState) => ({ ...prevState, autoTripIdPrefix: evt.target.value }))
```

Comment:
I would inline these methods where they're being used rather than define a variable for each. The same way you modified things to not use setStartTime, etc. anymore